src/P2P/PeerDiscovery.ts

```diff
@@ -0,0 +1,63 @@
+import PeerStore from "@polykey/PeerStore/PeerStore";
+import KeyManager from "@polykey/KeyManager";
+import MulticastBroadcaster from "@polykey/P2P/MulticastBroadcaster";
+import PeerInfo from "@polykey/PeerStore/PeerInfo";
+import fetch from 'node-fetch'
+
+class PeerDiscovery {
+  peerStore: PeerStore
+  keyManager: KeyManager
+  multicastBroadcaster: MulticastBroadcaster
+
+  constructor(peerStore: PeerStore, keyManager: KeyManager) {
+    this.peerStore = peerStore
+    this.keyManager = keyManager
+    this.multicastBroadcaster = new MulticastBroadcaster(this.peerStore, this.keyManager)
+  }
+
+  async start() {
+    await this.multicastBroadcaster.start()
+  }
+
+  async stop() {
+    await this.multicastBroadcaster.stop()
+  }
+
+  async findPubKey(pubKey: string): Promise<PeerInfo> {
+    return new Promise<PeerInfo>((resolve, reject) => {
+      this.multicastBroadcaster.requestPeerContact(pubKey)
+      this.multicastBroadcaster.on('found', (peerInfo: PeerInfo) => {
+        if (peerInfo.publicKey == pubKey) {
+          resolve(peerInfo)
+        }
+      })
+
+      this.multicastBroadcaster.on('timeout', (timedOutPubKey: string) => {
+        if (timedOutPubKey == pubKey) {
+          reject('The broadcaster stopped looking')
+        }
+      })
+    })
+  }
+
+  async findSocialUser(handle: string, service: 'username' | 'github' | 'twitter' | 'reddit' | 'hackernews'): Promise<PeerInfo> {
+    const url = `https://keybase.io/_/api/1.0/user/lookup.json?${service}=${handle}`
```

Comment:
Keybase is only one of the ways we can do this. I recommend using a strategy pattern for this. So we can extend it with more social discovery providers.

Reply:
Sure, I think we could provide and interface for this, so something like:
```
interface SocialDiscovery {
    findUser(handle, service): PubKey | null { ... }
}
```
Then the end user can create as many `SocialDiscovery` objects they want and perhaps pass them into `Polykey` class constructor.

Reply:
This is now implemented as an extra parameter in the constructor.